Yoast/yoast-components
# Implement redux provider in App.js

Adds a global redux provider to `App.js` for the test environment.

Hot reloading has been implemented for reloading reducers in `configureStore.js`.

`Redux devtools` have been implemented for easily debugging state changes. Press **Ctrl + H** to open the devtools.

## Test instructions
Replace the contents of `ContentAnalysis.js` with the following content:

Open the browser and press **Ctrl + H** to see the initial state. Now press the button on the `Content analysis` tab. You should see the state change in the devtools and a list on screen being updated.
```js
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { connect } from "react-redux";
import { updateSeoResult } from "../actions/contentAnalysis";

export const ContentAnalysisContainer = styled.div`
	min-height: 700px;
	width: 100%;
	background-color: white;
`;

/**
 * Returns the ContentAnalysis component.
 *
 * @returns {ReactElement} The ContentAnalysis component.
 */
class ContentAnalysis extends React.Component {
	updateSeoResult() {
		const keyword = "keyword";
		let length = 0;
		if( this.props.seo[ keyword ] ) {
			length = this.props.seo[ keyword ].length;
		}
		this.props.updateSeoResult( keyword, {
			id: `resultId${ length }`,
		} );
	}

	renderKeywordResults() {
		const keyword = "keyword";
		if( ! this.props.seo[ keyword ] ) {
			return null;
		}
		const results = this.props.seo[ keyword ];
		return results.map( ( result ) => {
			return (
				<li key={ result.id }>{ result.id }</li>
			);
		} );
	}

	render() {
		return (
			<ContentAnalysisContainer>
				<button onClick={ this.updateSeoResult.bind( this ) }>Add keyword result</button>
				<ol>
					{ this.renderKeywordResults() }
				</ol>
			</ContentAnalysisContainer>
		);
	};
}

ContentAnalysis.propTypes = {
	updateSeoResult: PropTypes.func,
	seo: PropTypes.object,
};

export default connect(
	state => ( {
		seo: state.contentAnalysis.seo,
	} ),
	{
		updateSeoResult,
	}
)( ContentAnalysis );
```

Fixes #236
